tests/unit/test_tensor.cpp

```diff
@@ -80,31 +85,50 @@ TEST(TensorTest, DynamicShape) {
 }
 
 TEST(TensorTest, CopyTo) {
-    {
-        deepworks::Tensor src_tensor({1, 3, 224, 224});
-        EXPECT_FALSE(src_tensor.empty());
-        for (size_t index = 0; index < 1 * 3 * 224 * 224; ++index) {
-            src_tensor.data()[index] = index;
-        }
-
-        deepworks::Tensor dst_tensor;
-        dst_tensor.allocate({1, 3, 224, 224});
-        src_tensor.copyTo(dst_tensor);
-
-        ASSERT_EQ(dst_tensor.shape(), src_tensor.shape());
-        ASSERT_EQ(dst_tensor.strides(), src_tensor.strides());
-        ASSERT_NE(dst_tensor.data(), src_tensor.data());
-        ASSERT_FALSE(dst_tensor.empty());
-        for (size_t index = 0; index < 1 * 3 * 224 * 224; ++index) {
-            ASSERT_EQ(dst_tensor.data()[index], index);
-        }
-
-        deepworks::Tensor non_empty_tensor({1, 3, 16, 16});
-        ASSERT_ANY_THROW(src_tensor.copyTo(non_empty_tensor));
+    deepworks::Tensor src_tensor({1, 3, 224, 224});
+    EXPECT_FALSE(src_tensor.empty());
+    for (size_t index = 0; index < 1 * 3 * 224 * 224; ++index) {
```

Comment:
Why not tensor.total()?

Reply:
Clean up tests a bit, but not dive deeper